cpp/src/arrow/util/future.h

```diff
@@ -691,20 +862,29 @@ struct Continue {
   }
 };
 
-template <typename T = detail::Empty>
+template <typename T = internal::Empty>
 util::optional<T> Break(T break_value = {}) {
   return util::optional<T>{std::move(break_value)};
 }
 
-template <typename T = detail::Empty>
+template <typename T = internal::Empty>
 using ControlFlow = util::optional<T>;
 
+template <typename T>
+void ForwardControlResult(const Result<ControlFlow<T>>& result, Future<T>& sink) {
```

Comment:
```suggestion
void ForwardControlResult(const Result<ControlFlow<T>>& result, Future<T>* sink) {
```

Reply:
At this point we're done with `break_fut` so I changed it to move `break_fut` and got rid of the `&`.